lisp/mastodon-profile.el

```diff
@@ -0,0 +1,154 @@
+
+;; mastodon.el is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with mastodon.el.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; mastodon-profile.el generates stream of users toots.
+;; To fix
+;; 1. Render Image at top of frame [x]
+;; 2. Get toot author [x]
+;; 3. Load more toots [x]
+;; Later
+;; 1. List followers [x]
+;; 2. List people they follow [x]
+;; 3. Option to follow
+;; 4. wheather they follow you or not
+;; 5. Show only Media
+
+;;; Code:
+
+(defun mastodon-profile--toot-json ()
+  "Get the next toot-json."
+  (interactive)
+  (mastodon-tl--property 'toot-json))
+
+(defun mastodon-profile--make-author-buffer (status)
+  "Take a STATUS and inserts a user account into a new buffer."
+  (let* ((id (mastodon-profile--field status 'id))
+         (acct (mastodon-profile--field status 'acct))
+         accounts/id/statuses
+         (url (mastodon-http--api
+               (concat "accounts/"
+                       (format "%s" id)
+                       "/statuses" )))
+         (buffer (concat "*mastodon-" acct  "*"))
+         (account (cdr(assoc 'account status)))
+         (note (mastodon-profile--field status 'note))
+         (json (mastodon-http--get-json url)))
+    (with-output-to-temp-buffer buffer
+      (switch-to-buffer buffer)
+      (mastodon-mode)
+      (setq mastodon-tl--buffer-spec
+            `(buffer-name ,buffer
+                          endpoint ,(format "accounts/%s/statuses" id)
+                          update-function
+                          ,'mastodon-tl--timeline json))
+      (let ((inhibit-read-only t))
+        "\n"
+        (mastodon-profile--image-from-status account)
+        "\n"
+        (propertize (mastodon-profile--field status 'display_name
+                                             'face 'mastodon-display-name-face))
+        "\n"
+        (propertize acct
+                    'face 'default)
+        "\n ------------\n"
+        (mastodon-tl--render-text note status)
+        (mastodon-tl--set-face
+         (concat " ------------\n"
+                 "     TOOTS   \n"
+                 " ------------\n")
+         'success)
+        (mastodon-tl--timeline json)))
+    (mastodon-tl--goto-next-toot)))
+
+(defun mastodon-profile--get-next-author ()
+  "Jump to the next author and generate a buffer."
+  (interactive)
+  (mastodon-profile--make-author-buffer (mastodon-profile--toot-json)))
+
+(defun mastodon-profile--image-from-status (status)
+  "Generate an image from a STATUS."
+  (let ((url (cdr(assoc 'avatar_static status))))
+    (unless (equal url "/avatars/original/missing.png")
+      (mastodon-media--get-media-link-rendering url))))
+
+(defun mastodon-profile--field (status field)
```

Comment:
Neither the function name nor the docstring gives me an idea of what this does.

I think you should work with the account json object and not the toot.  Then you can lose the inner assoc and this becomes a simple concatenation of `cdr` and `assoc` which doesn't really add any abstraction so we could get rid of it altogether.